Joyent driver broken on develop when provisioning instances

When trying to spin up a VM on Joyent using salt-cloud, the following stacktrace occurs:
```
[DEBUG   ] Joyent Response Status Code: 200
[ERROR   ] There was a profile error: dictionary changed size during iteration
Traceback (most recent call last):
  File "/root/SaltStack/salt/salt/cloud/cli.py", line 255, in run
    self.config.get('names')
  File "/root/SaltStack/salt/salt/cloud/__init__.py", line 1378, in run_profile
    ret[name] = self.create(vm_)
  File "/root/SaltStack/salt/salt/cloud/__init__.py", line 1248, in create
    output = self.clouds[func](vm_)
  File "/root/SaltStack/salt/salt/cloud/clouds/joyent.py", line 307, in create
    query_instance(vm_)
  File "/root/SaltStack/salt/salt/cloud/clouds/joyent.py", line 227, in query_instance
    'wait_for_ip_interval_multiplier', vm_, __opts__, default=1),
  File "/root/SaltStack/salt/salt/utils/cloud.py", line 2012, in wait_for_ip
    data = update_callback(*update_args, **update_kwargs)
  File "/root/SaltStack/salt/salt/cloud/clouds/joyent.py", line 198, in _query_ip_address
    data = show_instance(vm_['name'], call='action')
  File "/root/SaltStack/salt/salt/cloud/clouds/joyent.py", line 633, in show_instance
    node = get_node(name)
  File "/root/SaltStack/salt/salt/cloud/clouds/joyent.py", line 614, in get_node
    nodes = list_nodes()
  File "/root/SaltStack/salt/salt/cloud/clouds/joyent.py", line 724, in list_nodes
    ret[node['name']] = reformat_node(item=node, full=full)
  File "/root/SaltStack/salt/salt/cloud/clouds/joyent.py", line 690, in reformat_node
    for key in six.iterkeys(item):  # iterate over a copy of the keys
RuntimeError: dictionary changed size during iteration
```
The instance does get created on the Joyent account, but since the `salt-cloud -p` command stacktraces, the VM never gets bootstrapped with salt.

Note: salt-cloud works just fine with the joyent driver on the 2015.2 branch.

ping @techhat

I just tried this again and I am now getting a different error:
```
[ERROR   ] There was a profile error: Body must be empty for "GET" request
Traceback (most recent call last):
  File "/root/SaltStack/salt/salt/cloud/cli.py", line 255, in run
    self.config.get('names')
  File "/root/SaltStack/salt/salt/cloud/__init__.py", line 1414, in run_profile
    ret[name] = self.create(vm_)
  File "/root/SaltStack/salt/salt/cloud/__init__.py", line 1256, in create
    output = self.clouds[func](vm_)
  File "/root/SaltStack/salt/salt/cloud/clouds/joyent.py", line 278, in create
    'image': get_image(vm_),
  File "/root/SaltStack/salt/salt/cloud/clouds/joyent.py", line 139, in get_image
    images = avail_images()
  File "/root/SaltStack/salt/salt/cloud/clouds/joyent.py", line 820, in avail_images
    rcode, data = query(command='my/images')
  File "/root/SaltStack/salt/salt/cloud/clouds/joyent.py", line 1073, in query
    opts=__opts__,
  File "/root/SaltStack/salt/salt/utils/http.py", line 394, in query
    **req_kwargs
  File "/usr/local/lib/python2.7/dist-packages/tornado/httpclient.py", line 96, in fetch
    self._async_client.fetch, request, **kwargs))
  File "/usr/local/lib/python2.7/dist-packages/tornado/ioloop.py", line 418, in run_sync
    return future_cell[0].result()
  File "/usr/local/lib/python2.7/dist-packages/tornado/concurrent.py", line 111, in result
    raise self._exception
AssertionError: Body must be empty for "GET" request
```
Versions report:
```
# salt-cloud --versions
Salt Version:
            Salt: 2015.5.0-1583-gd299475

Dependency Versions:
 Apache Libcloud: 0.17.0
          Jinja2: 2.7.3
        M2Crypto: 0.21.1
            Mako: 0.9.1
          PyYAML: 3.11
           PyZMQ: 14.0.1
          Python: 2.7.6 (default, Mar 22 2014, 22:59:56)
            RAET: Not Installed
         Tornado: 4.0.2
             ZMQ: 4.0.4
           ioflo: Not Installed
         libnacl: Not Installed
    msgpack-pure: Not Installed
  msgpack-python: 0.4.4
        pycrypto: 2.6.1

System Versions:
            dist: Ubuntu 14.04 trusty
         machine: x86_64
         release: 3.13.0-30-generic
          system: Ubuntu 14.04 trusty
```

--- a/salt/cloud/clouds/joyent.py
+++ b/salt/cloud/clouds/joyent.py
@@ -690,10 +690,14 @@ def reformat_node(item=None, full=False):
             item[key] = None
 
     # remove all the extra key value pairs to provide a brief listing
+    to_del = []
     if not full:
         for key in six.iterkeys(item):  # iterate over a copy of the keys
             if key not in desired_keys:
-                del item[key]
+                to_del.append(key)
+
+    for key in to_del:
+        del item[key]
 
     if 'state' in item:
         item['state'] = joyent_node_state(item['state'])

